### ISSUE TITLE
PollSet filters out some events

### ISSUE THREAD

**Issue by micheleselea** [pocoproject/poco#4194]
on windows 11 (but I think all windows version that use wepoll.c) I found a bad behavior if the internal wepoll request do not return one of EPOLLIN, EPOLLOUT, or EPOLLERR
because in PollSet we check that return code in this code
```
if (_events[i].events & EPOLLIN)
				result[it->second.first] |= PollSet::POLL_READ;
if (_events[i].events & EPOLLOUT)
				result[it->second.first] |= PollSet::POLL_WRITE;
if (_events[i].events & EPOLLERR)
				result[it->second.first] |= PollSet::POLL_ERROR;
```
there's a condition where the wepoll can return 
```
 if (afd_events & AFD_POLL_RECEIVE_EXPEDITED)
   epoll_events |= EPOLLPRI | EPOLLRDBAND;
```
or
```
if (afd_events & AFD_POLL_ABORT)
  epoll_events |= EPOLLHUP;
```

In that situation the PollSet check do not intercept any valid result, but the wepoll do not wait on timeout so a socket reactor can end up in a while(1) situation. My specific case was EPOLLHUP that is not catched (I think the correct way here is a POLL_ERROR, but probably even  EPOLLPRI | EPOLLRDBAND could be hadled with a POLL_READ

**Comment by micheleselea** [comment]
The way I reproduce it was to accept an FTP connection and connecting with a client configured with FTPS Implicit connection. Don't know if happen only with my specific client

**Comment by aleks-f** [comment]
@micheleselea which branch and is this the same issue as #3773 ?

To facilitate a faster trunaround, a [SSCCE](http://sscce.org) is better than prose and code snippets

**Comment by micheleselea** [comment]
It's the same result (while(1))
https://github.com/pocoproject/poco/issues/3773
but the problem is generated by another issue: in 3773 is the opening listen port that if silently connected to can make wepoll do a while(1) because it's out of PollSet control.
In this case the problem is on a socket controlled by PollSet but that result in EPOLLHUP event that is not handled by PollSet.
I try to put together a working sample

**Comment by micheleselea** [comment]
looking this function
```
static inline uint32_t sock__afd_events_to_epoll_events(DWORD afd_events) {
  uint32_t epoll_events = 0;

  if (afd_events & (AFD_POLL_RECEIVE | AFD_POLL_ACCEPT))
    epoll_events |= EPOLLIN | EPOLLRDNORM;
  if (afd_events & AFD_POLL_RECEIVE_EXPEDITED)
    epoll_events |= EPOLLPRI | EPOLLRDBAND;
  if (afd_events & AFD_POLL_SEND)
    epoll_events |= EPOLLOUT | EPOLLWRNORM | EPOLLWRBAND;
  if (afd_events & AFD_POLL_DISCONNECT)
    epoll_events |= EPOLLIN | EPOLLRDNORM | EPOLLRDHUP;
  if (afd_events & AFD_POLL_ABORT)
    epoll_events |= EPOLLHUP;
  if (afd_events & AFD_POLL_CONNECT_FAIL)
    /* Linux reports all these events after connect() has failed. */
    epoll_events |=
        EPOLLIN | EPOLLOUT | EPOLLERR | EPOLLRDNORM | EPOLLWRNORM | EPOLLRDHUP;

  return epoll_events;
}
```
I think the events to catch are
POLL_READ -> EPOLLIN,EPOLLHUP.EPOLLPRI
POLL_WRITE-> EPOLLOUT
POLL_ERROR-> EPOLLERR

**Comment by micheleselea** [comment]
@aleks-f to easy simulate the problem end obtain EPOLLHUP on wepoll open a socket reactor and connect with FileZilla Client with a "Require Implicit FTP over TLS" connection
